python/src/scipp/plot/plot.py

```diff
@@ -2,9 +2,89 @@
 # Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
 # @author Neil Vaytet
 
+from typing import List, Dict
+
 # Scipp imports
-from .._scipp import core as sc
+from scipp.plot.prepare_collapse import prepare_collapse
+from scipp.plot.plot_request import PlotRequest
 from .sciplot import SciPlot
+from .tools import get_line_param
+from .._scipp import core as sc
+
+
+def tiled_plot(scipp_objs: List,
+               collapse=None,
+               projection=None,
+               axes=None,
+               color=None,
+               marker=None,
+               linestyle=None,
+               linewidth=None,
+               bins=None,
+               **kwargs):
+    if isinstance(scipp_objs, list) or isinstance(scipp_objs, set):
+        packed_objs = {}
+        for obj in scipp_objs:
+            obj_dict = _pack_into_dict(obj)
+            if not obj_dict:
+                # Nothing to plot
+                continue
+
+            # We need to be careful of duplicates -
+            # keep trying keys until we find a unique name
+            assert len(obj_dict) == 1, "Duplicate name handling requires" \
+                                       " _pack_into_dict to return one val"
+            original_key = str(next(iter(obj_dict)))
+            key = original_key
+            i = 1
+            while key in packed_objs:
+                key = f"{original_key}_{i}"
```

Comment:
Tthis would not be needed if you are accepting only a `Dataset` (or a `dict`) instead of a list